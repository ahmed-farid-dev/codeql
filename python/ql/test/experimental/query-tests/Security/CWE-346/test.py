from flask import Response, request, Flask

@app.route("/flask_Response")
def flask_Response():
    rfs_header = request.args["rfs_header"]
    response = Response()
    response.headers['Access-Control-Allow-Origin'] = rfs_header
    response.headers['Access-Control-Allow-Credentials'] = "true"
    return response
