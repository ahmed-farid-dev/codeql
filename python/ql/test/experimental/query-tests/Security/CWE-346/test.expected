| test.py:7:55:7:64 | ControlFlowNode for rfs_header | test.py:5:18:5:24 | ControlFlowNode for request | test.py:7:55:7:64 | ControlFlowNode for rfs_header | CORS header is being set using user controlled value $@. | test.py:5:18:5:24 | ControlFlowNode for request | user-provided value |
